### PR TITLE
BAK-1305 Create and populate metadata_taxonomy

### PR DESCRIPTION
**Pull request by diversemix**
The first of several PRs to populate and use the taxonomy on ingestion.

### CODE REVIEW

File: app/data_migrations/populate_taxonomy.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+def populate_taxonomy(db: Session) -> None:
+    """Populates the taxonomy from the data."""
+
+    if has_rows(db, MetadataTaxonomy):
+        return
+
+    db.add(MetadataTaxonomy(
+        name="default", 
+        description="CCLW loaded values", 
+        valid_metadata=get_default_taxonomy()))
```
Comment (joel-wright) [inline app/data_migrations/populate_taxonomy.py:110]: we could load the initial data from the json files - doing it from the existing db rows will be a pain when we remove the old metadata tables...

Comment (joel-wright) [reply app/data_migrations/populate_taxonomy.py:110]: okay, may have misread it - brb

File: app/data_migrations/populate_taxonomy.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    if "." not in dotted_key:
+        return obj[dotted_key]
+    keys = dotted_key.split(".", 1)
+    return dot_dref(obj[keys[0]], keys[1])
+    
+
+def load_metadata_type(filename: str, key_path: str) -> Sequence[str]:
+    with open(filename) as file:
+        data = json.load(file)
+    return [dot_dref(obj, key_path) for obj in data]
```
Comment (diversemix) [inline app/data_migrations/populate_taxonomy.py:85]: @joel-wright This is the function that loads the metadata from the data files :smile:

**Review by joel-wright — COMMENTED** [review]
LGTM

**Review by joel-wright — APPROVED** [review]

**Pull request merged**